### ISSUE TITLE
Multi resolve and false positive E0308 expected `Span`, found `Option<Span>`

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8220]
## Environment

* **IntelliJ Rust plugin version:** 0.4.162.4282-213-nightly
* **Rust toolchain version:** 1.58.0-nightly (547a6ffee 2021-10-21) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2021.3 Ultimate Edition (IU-213.5744.223)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
mismatched types [E0308] expected `Span`, found `Option<Span>`
<img width="858" alt="изображение" src="https://user-images.githubusercontent.com/49211026/144583192-70f96e22-d6cd-4db4-8020-3fd68c23c79a.png">

All exceptions can be fixed by adding `use syn::spanned::Spanned;`, but plugin detects two variants to import, so main problem is resolve. 
<img width="449" alt="изображение" src="https://user-images.githubusercontent.com/49211026/144592500-2958d4e7-f6fb-4972-9927-121b07b92ce8.png">
(Without adding import code is working)

## Steps to reproduce
main.rs:
```rust
use proc_macro2::Span;
use syn::{parse::Error, Ident, Lit, Meta};

fn parse_attrs(field: &syn::Field) -> Result<(), Error> {
    for attr in field.attrs.iter() {
        let meta = attr.parse_meta()?;
        if let Meta::NameValue(value) = meta {
            if value.path.is_ident("some_value") {
                if let Lit::Int(val) = value.lit {
                    match val.base10_parse::<usize>() {
                        Ok(_value) => {
                            //do things with value
                        }
                        Err(err) => {
                            return Err(Error::new(
                                if let Some(ref ident) = field.ident { // error e0308
                                    ident.span()   // unresolved reference
                                } else {
                                    Span::call_site()
                                },
                                format!("fail parsing int literal as usize [{}]", err),
                            ));
                        }
                    }
                } else {
                    return Err(Error::new(
                        if let Some(ref ident) = field.ident {   // error e0308
                            ident.span()
                        } else {
                            Span::call_site()
                        },
                        "must be int literal",
                    ));
                }
            }
        }
    }
    Ok(())
}

fn main() {}

```

cargo.toml
```toml


[lib]
proc-macro = true

[dependencies]
proc-macro2 = "1.0"
quote = "1.0"
syn = "1.0"
proc-macro-error = "1.0.4"
log = "0.4.14"
```
